subworkflows/canine_tumor_only_variant_filter_module.cwl

```diff
@@ -0,0 +1,80 @@
+cwlVersion: v1.2
+class: Workflow
+id: canine_tumor_only_variant_filter_module
+doc: "Port of Canine Tumor Only Variant Filter Module"
+
+requirements:
+- class: ScatterFeatureRequirement
+- class: StepInputExpressionRequirement
+- class: MultipleInputFeatureRequirement
+- class: InlineJavascriptRequirement
+
+inputs:
+  input_vcf: { type: 'File', doc: "VCF file to annotate." }
+  output_basename: { type: 'string', doc: "String to use as base for output filenames." }
+  disable_workflow: { type: 'boolean?', doc: "For when this workflow is wrapped into a larger workflow, you can use this value in the when statement to toggle the running of this workflow." }
+
+  # Resource Control
+  bcftools_ram: { type: 'int?', doc: "Maximum GB of RAM to allocate to BCFtools." }
+  bcftools_cpu: { type: 'int?', doc: "Number of CPUs to allocate to BCFtools." }
+
+outputs:
+  filtered_vcf: { type: 'File', outputSource: bcftools_concat_sort_index/output }
+
+steps:
+  bcftools_view_view_index_filtered:
+    run: ../tools/bcftools_view_view_index.cwl
+    in:
+      input_vcf:  input_vcf
+      include:
+        source: disable_workflow # hiding this here because I hate cavatica
```

Comment:
So pretty much I need to sink the disable_workflow input into a step otherwise cavatica will give me the "graph not connected" error and refuse to run the app. Originally I was putting it in its own clearly defined expression tool but apparently cavatica didn't like that either. So I got rid of the expression tool and now I'm doing what I have here. I am sourcing the disable_workflow variable someplace where it doesn't matter. Here you can see that regardless of what I put in the source, the subsequent valueFrom statement will still always return its string.

Truthfully I've written this exact comment in every subworkflow that can be disabled. If you would humor me for a second, I'd rather not update all the PRs so if you approve this, I'll make sure to remove this and all other lines in the final PR where I'm going to make the main workflow.

Reply:
Makes sense about updating it in the final PR.